tensorflow/core/kernels/data/batch_dataset_op_test.cc

```diff
@@ -51,677 +52,973 @@ class BatchDatasetOpTest : public DatasetOpsTestBase {
   }
 };
 
-struct RangeDatasetParam {
-  int64 start;
-  int64 end;
-  int64 step;
-};
+class BatchDatasetParams : public DatasetParams {
+ public:
+  BatchDatasetParams(int64 start, int64 stop, int64 step, int64 batch_size,
+                     bool drop_remainder, bool parallel_copy,
+                     DataTypeVector output_dtypes,
+                     std::vector<PartialTensorShape> output_shapes,
+                     string node_name)
+      : DatasetParams(std::move(output_dtypes), std::move(output_shapes),
+                      std::move(node_name)),
+        range_dataset_params(start, stop, step, {DT_INT64},
+                             {PartialTensorShape({})}, ""),
+        batch_size(CreateTensor<int64>(TensorShape({}), {batch_size})),
+        drop_remainder(CreateTensor<bool>(TensorShape({}), {drop_remainder})),
+        parallel_copy(parallel_copy) {}
+
+  Status MakeInputs(gtl::InlinedVector<TensorValue, 4>* inputs) override {
+    if (input_dataset.NumElements() == 0 ||
+        input_dataset.dtype() != DT_VARIANT) {
+      return tensorflow::errors::Internal(
+          "The input dataset is not populated as the dataset tensor yet.");
+    }
+    *inputs = {TensorValue(&input_dataset), TensorValue(&batch_size),
+               TensorValue(&drop_remainder)};
+    return Status::OK();
+  }
 
-struct TestCase {
-  RangeDatasetParam range_dataset_param;
+  RangeDatasetParams range_dataset_params;  // Used to create the input dataset.
+  Tensor input_dataset;
   Tensor batch_size;
   Tensor drop_remainder;
   bool parallel_copy;
-  std::vector<Tensor> expected_outputs;
-  DataTypeVector expected_output_dtypes;
-  std::vector<PartialTensorShape> expected_output_shapes;
-  int64 expected_cardinality;
-  std::vector<int> breakpoints;
 };
 
 // Test Case 1: test BatchDatasetV2 with `drop_remainder` = false and a batch
 // size that can evenly split the input dataset.
-TestCase TestCase1() {
-  return {/*range_data_param*/ {0, 12, 1},
-          /*batch_size*/
-          CreateTensor<int64>(TensorShape({}), {4}),
-          /*drop_remainder*/
-          CreateTensor<bool>(TensorShape({}), {false}),
-          /*parallel_copy*/ true,
-          /*expected_outputs*/
-          {CreateTensor<int64>(TensorShape({4}), {0, 1, 2, 3}),
-           CreateTensor<int64>(TensorShape({4}), {4, 5, 6, 7}),
-           CreateTensor<int64>(TensorShape({4}), {8, 9, 10, 11})},
-          /*expected_output_dtypes*/ {DT_INT64},
-          /*expected_output_shapes*/ {PartialTensorShape({4})},
-          /*expected_cardinality*/ 3,
-          /*breakpoints*/ {0, 1, 5}};
+BatchDatasetParams BatchDataset1() {
+  return {/*start=*/0,
+          /*stop=*/12,
+          /*step=*/1,
```

Comment:
the step and start is 1 and 0 for all the test cases so the test case should just specify `num_input_elements` and the implementation should create an input dataset that matches that

Reply:
They are changed to `num_input_elements` [here](https://github.com/tensorflow/tensorflow/pull/31344/commits/ec110b9a308f2c783c4a3ac0967377770c1f8ba7#diff-446881c338865246450f993442bc7ec6R51).